Userland/Services/WebDriver/Client.cpp

```diff
@@ -332,7 +331,18 @@ Web::WebDriver::Response Client::switch_to_window(Web::WebDriver::Parameters par
 {
     dbgln_if(WEBDRIVER_DEBUG, "Handling POST /session/<session_id>/window");
     auto* session = TRY(find_session_with_id(parameters[0]));
-    return session->web_content_connection().switch_to_window(payload);
+
+    if (!payload.is_object())
+        return Web::WebDriver::Error::from_code(Web::WebDriver::ErrorCode::InvalidArgument, "Payload is not a JSON object");
+
+    // 1. Let handle be the result of getting the property "handle" from the parameters argument.
+    auto handle = payload.as_object().get("handle"sv);
+
+    // 2. If handle is undefined, return error with error code invalid argument.
+    if (!handle.has_value())
+        return Web::WebDriver::Error::from_code(Web::WebDriver::ErrorCode::InvalidArgument, "No property called 'handle' present");
+
+    return session->switch_to_window(MUST(String::from_deprecated_string(handle->as_string())));
```

Comment:
`MUST` -> `release_value_but_fixme_should_propagate_errors`

Reply:
You can avoid this altogether actually. `switch_to_window` can be made to take a `StringView` parameter, as we can do lookups into hash maps keyed by `String` with a `StringView` (we call it "hash compatible").

Reply:
This isn't quite resolved - we should not use `MUST` to hide errors that should instead be propagated eventually. But the point of changing `switch_to_window` to take a `StringView` is that this invocation can now just be:
```c++
return session->switch_to_window(handle->as_string());
```
so that we don't have to worry about error propagation here

Reply:
sorry, yes, I have to invest some time in learning how to deal with strings :)